packages/react-components/src/components/MessageThread.tsx

```diff
@@ -361,14 +374,107 @@ const memoizeAllMessages = memoizeFnAll(
+        if (message.messageType === 'blocked') {
+          messageProps.messageContainerStyle = blockedMessageStyle;
+        }
+      }
+
+      const chatMessageComponent =
+        onRenderMessage === undefined
+          ? defaultChatMessageRenderer(messageProps)
+          : onRenderMessage(messageProps, defaultChatMessageRenderer);
+
+      const personaOptions: IPersona = {
+        hidePersonaDetails: true,
+        size: PersonaSize.size32,
+        text: message.senderDisplayName,
+        showOverflowTooltip: false
+      };
+
+      const chatItemMessageStyle =
+        (message.mine ? styles?.myChatItemMessageContainer : styles?.chatItemMessageContainer) ||
+        defaultChatItemMessageContainer(shouldOverlapAvatarAndMessage);
+
+      const chatGutterStyles =
+        message.attached === 'top' || message.attached === false ? gutterWithAvatar : gutterWithHiddenAvatar;
+
+      return {
+        gutter: {
+          styles: chatGutterStyles,
+          content: message.mine ? (
+            ''
+          ) : onRenderAvatar ? (
+            onRenderAvatar(message.senderId ?? '', personaOptions)
+          ) : (
+            <Persona {...personaOptions} />
+          )
+        },
+        contentPosition: message.mine ? 'end' : 'start',
+        message: {
+          styles: chatItemMessageStyle,
+          content: (
+            <Flex hAlign={message.mine ? 'end' : undefined} vAlign="end">
+              {chatMessageComponent}
+              <div
+                className={mergeStyles(
+                  messageStatusContainerStyle(message.mine ?? false),
+                  styles?.messageStatusContainer ? styles.messageStatusContainer(message.mine ?? false) : ''
+                )}
+              >
+                {showMessageStatus && statusToRender ? (
+                  onRenderMessageStatus ? (
+                    onRenderMessageStatus({ status: statusToRender })
+                  ) : (
+                    defaultStatusRenderer(message, statusToRender, participantCount ?? 0, readCount ?? 0)
+                  )
+                ) : (
+                  <div className={mergeStyles(noMessageStatusStyle)} />
+                )}
+              </div>
+            </Flex>
+          )
+        },
+        attached: message.attached,
+        key: _messageKey
+      };
+    }
+
     switch (message.messageType) {
+      // case 'blocked':
       case 'chat': {
         const myChatMessageStyle =
           message.status === 'failed'
             ? styles?.failedMyChatMessageContainer ?? styles?.myChatMessageContainer ?? FailedMyChatMessageContainer
             : styles?.myChatMessageContainer ?? defaultMyChatMessageContainer;
         const chatMessageStyle = styles?.chatMessageContainer ?? defaultChatMessageContainer(theme);
-        messageProps.messageContainerStyle = message.mine ? myChatMessageStyle : chatMessageStyle;
+        /* @conditional-compile-remove(data-loss-prevention) */
```

Comment:
Why do we have a @conditional-compile-remove(data-loss-prevention)  inside of the 'chat' condition?

Reply:
Because the conditional-compile-remove does not work for switch statement case, so extracting out the logic to a `if` before the switch-statement. (
By adding the conditional compile here so the `if` and `case chat` will almost be identical and can more easily be removed when shipping to stable.